File: Python/0050_powx_n/0050_powx_n.py
class Solution:
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        temp = n
        if temp == 0:
            return 1
        elif temp < 0:
            temp = -temp
            x = 1.0 / x;
        elif temp == 1:
            return x
        return self.myPow(x * x, temp >> 1) * (1 if n % 2 == 0 else x)

File: Java/0050_powx_n/0050_powx_n.java
class Solution {
    public double myPow(double x, int n) {
        if (n == 0)
            return 1;
        else if (n < 0) 
            if (n == Integer.MIN_VALUE) {
                n = Integer.MAX_VALUE;
                if (x < 0)
                    x = -x;
            }
            else
                n = -n;
            x = 1 / x;
        }
        return myPow(x * x, n >> 1) * (n % 2 == 0 ? 1 : x);
    }
}